clash-cores/src/Clash/Cores/Xilinx/VIO.hs

```diff
@@ -75,7 +75,7 @@ instance VIO dom a o => VIO dom (Signal dom i -> a) o where
 --
 -- @
 -- someProbe :: 'KnownDomain' dom => 'Clock' dom -> 'Signal' dom 'Bit' -> 'Signal' dom ('Unsigned' 8) -> 'Signal' dom ('Bool', 'Maybe' ('Signed' 8))
--- someProbe = 'vioProbe' ('False', 'Nothing')
+-- someProbe = 'vioProbe' ("in_b" :> "in_u8" :> Nil) ("out_b" :> "out_mu8" :> Nil) ('False', 'Nothing')
```

Comment:
Clash cannot translate lists to `Expr`/`HWType` AFAIK. But also; there is no reason we couldn't add length restrictions using a type family in the future, right? So I'd argue in favor of `Vec`.

Reply:
> Some other useful extension may be to support optional naming via providing less elements then necessary, i.e., if less names are given than probes are available, the names are taken until used up, and then defaults are used instead.

We discussed this in the office and decided against it: Clash does this for `Synthesize` pragmas, and it can be incredibly frustrating as this can go unnoticed for a bit. E.g., for the `Synthesize` pragmas we were missing a port name "in the middle" at some point, which caused a lot of port names to be "shifted by one" causing wrongly connected pins /  broken hardware.

I'm not entirely against special casing `Nil` for auto generation, but then again: do you really want to use a VIO without assigning names?

Reply:
> do you really want to use a VIO without assigning names?

Hidde says: **NO**.